### PR TITLE
Introduced TicketBai invoice models

### PR DESCRIPTION
**Pull request by abdallahbeshi**
## Description

Introduced TicketBai invoice models.

image

## Type of change
- [ ] Bug fix.
- [x] Feature.
- [ ] Consolidation.

## Checklist
- [ ] Is breaking change.
- [ ] Documentation updated.
- [ ] Tests included (please specify cases).

### CODE REVIEW

File: src/TicketBai/Mews.Fiscalizations.TicketBai/Model/Invoice/CorrectingInvoiceAmount.cs [hunk 1]
```diff
@@ -0,0 +1,18 @@
+﻿namespace Mews.Fiscalizations.TicketBai.Model
+{
+    public sealed class CorrectingInvoiceAmount
+    {
+        public CorrectingInvoiceAmount(decimal amount, decimal fee, decimal surcharge)
```
Comment (KaliCZ) [inline src/TicketBai/Mews.Fiscalizations.TicketBai/Model/Invoice/CorrectingInvoiceAmount.cs:5]: `Corrective`

File: src/TicketBai/Mews.Fiscalizations.TicketBai/Model/Invoice/CorrectingInvoice.cs [hunk 2]
```diff
@@ -0,0 +1,20 @@
+﻿using FuncSharp;
+
+namespace Mews.Fiscalizations.TicketBai.Model
+{
+    public sealed class CorrectingInvoice
```
Comment (KaliCZ) [inline src/TicketBai/Mews.Fiscalizations.TicketBai/Model/Invoice/CorrectingInvoice.cs:5]: `Corrective`

File: src/TicketBai/Mews.Fiscalizations.TicketBai/Model/Invoice/InvoiceHeader.cs [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+    public sealed class InvoiceHeader
+    {
+        public InvoiceHeader(
+            String1To20 number,
+            DateTime issueDate,
+            DateTime issueDateTime,
+            bool isSimplified = false,
+            bool issuedInSubstitutionOfSimplifiedInvoice = false,
+            String1To20 series = null,
+            CorrectingInvoice correctingInvoice = null,
```
Comment (KaliCZ) [inline src/TicketBai/Mews.Fiscalizations.TicketBai/Model/Invoice/InvoiceHeader.cs:18]: I don't understand. So I'm submitting invoice 20 and I'm already saying that there's gonna be an invoice 21 correcting this stuff?

Comment (KaliCZ) [reply src/TicketBai/Mews.Fiscalizations.TicketBai/Model/Invoice/InvoiceHeader.cs:18]: With juts common sense, all I could think of is that CorrectingInvoice  is actually the data about `how much money did we correct with this`. Which would mean it's more like Data and this is the correcting invoice. But hey, what do I know. First time I hear of this. We could try asking someone.

**Review by KaliCZ — APPROVED** [review]
BTW, you do expect that I will make you use coproducts later on, right?

File: src/TicketBai/Mews.Fiscalizations.TicketBai/Model/Invoice/CorrectingInvoice.cs (hunk 2 above)
Comment (KaliCZ) [reply src/TicketBai/Mews.Fiscalizations.TicketBai/Model/Invoice/CorrectingInvoice.cs:5]: You know what? Let's leave this until we figure out what it actually is 😄

File: src/TicketBai/Mews.Fiscalizations.TicketBai/Model/Invoice/CorrectingInvoiceAmount.cs (hunk 1 above)
Comment (KaliCZ) [reply src/TicketBai/Mews.Fiscalizations.TicketBai/Model/Invoice/CorrectingInvoiceAmount.cs:5]: You know what? Let's leave this until we figure out what it actually is 😄

File: src/TicketBai/Mews.Fiscalizations.TicketBai/Model/Invoice/InvoiceHeader.cs (hunk 3 above)
Comment (abdallahbeshi) [reply src/TicketBai/Mews.Fiscalizations.TicketBai/Model/Invoice/InvoiceHeader.cs:18]: Yea exactly, I'm definitely gonna visit this again later.
I will dig deeper into the docs and examples to figure it out.

**Pull request merged**